perfkitbenchmarker/linux_benchmarks/cloud_spanner_ycsb_benchmark.py

```diff
@@ -0,0 +1,204 @@
+                    'The schema DDL for the Cloud Spanner database.')
+
+
+def GetConfig(user_config):
+  config = configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+  if FLAGS['ycsb_client_vms'].present:
+    config['vm_groups']['default']['vm_count'] = FLAGS.ycsb_client_vms
+  return config
+
+
+def CheckPrerequisites(benchmark_config):
+  for scope in REQUIRED_SCOPES:
+    if scope not in FLAGS.gcloud_scopes:
+      raise ValueError('Scope {0} required.'.format(scope))
+
+
+def Prepare(benchmark_spec):
+  """Prepare the virtual machines to run cloud spanner benchmarks.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  benchmark_spec.always_call_cleanup = True
+
+  benchmark_spec.spanner_instance = gcp_spanner.GcpSpannerInstance(
+      FLAGS.cloud_spanner_instance, FLAGS.cloud_spanner_desp,
+      FLAGS.cloud_spanner_nodes, FLAGS.cloud_spanner_config,
+      FLAGS.cloud_spanner_database, FLAGS.cloud_spanner_ddl)
+  benchmark_spec.spanner_instance.Delete()
+  benchmark_spec.spanner_instance.Create()
+  if not benchmark_spec.spanner_instance._Exists():
+    logging.warning('Failed to create Cloud Spanner instance and database.')
+    benchmark_spec.spanner_instance.Delete()
+
+  default_ycsb_tar_url = ycsb.YCSB_TAR_URL
+
+  # TODO: figure out a less hacky way to override.
+  # Override so that we only need to download the required binding.
+  if FLAGS.cloud_spanner_custom_ycsb_release:
+    ycsb.YCSB_TAR_URL = FLAGS.cloud_spanner_custom_ycsb_release
+  else:
+    ycsb.YCSB_TAR_URL = YCSB_BINDING_TAR_URL
+
+  logging.warning('YCSB tar url: ' + ycsb.YCSB_TAR_URL)
+
+  vms = benchmark_spec.vms
+
+  # Install required packages and copy credential files
+  vm_util.RunThreaded(_Install, vms)
+
+  # Restore YCSB_TAR_URL
+  ycsb.YCSB_TAR_URL = default_ycsb_tar_url
+  benchmark_spec.executor = ycsb.YCSBExecutor('cloudspanner')
+
+
+def Run(benchmark_spec):
+  """Spawn YCSB and gather the results.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+
+  Returns:
+    A list of sample.Sample instances.
+  """
+  vms = benchmark_spec.vms
+  run_kwargs = {
+      'table': FLAGS.cloud_spanner_table,
+      'zeropadding': FLAGS.cloud_spanner_zeropadding,
+      'cloudspanner.instance': FLAGS.cloud_spanner_instance,
+      'cloudspanner.database': FLAGS.cloud_spanner_database,
+      'cloudspanner.readmode': FLAGS.cloud_spanner_readmode,
+      'cloudspanner.boundedstaleness': FLAGS.cloud_spanner_boundedstaleness,
+      'cloudspanner.batchinserts': FLAGS.cloud_spanner_batchinserts,
+  }
+  if FLAGS.cloud_spanner_host:
+    run_kwargs['cloudspanner.host'] = FLAGS.cloud_spanner_host
+
+  load_kwargs = run_kwargs.copy()
+  if not FLAGS['ycsb_preload_threads'].present:
+    load_kwargs['threads'] = 1
+  samples = list(benchmark_spec.executor.LoadAndRun(
```

Comment:
Yes, but not in this pull request. We are going to determine a common set of properties shared by cloud spanner, bigtable, datastore, and then we'll update all the benchmarks.

Added a TODO